app/src/main/java/agency/five/codebase/android/movieapp/ui/moviedetails/MovieDetailsScreen.kt

```diff
@@ -0,0 +1,214 @@
+import androidx.compose.foundation.lazy.items
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.graphics.Color
+
+private val movieDetailsMapper: MovieDetailsMapper = MovieDetailsMapperImpl()
+
+val movieDetailsViewState = movieDetailsMapper.toMovieDetailsViewState(MoviesMock.getMovieDetails())
+
+@Composable
+fun MovieDetailsRoute(
+) {
+    val details by remember { mutableStateOf(movieDetailsViewState) }
+    MovieDetailsScreen(
+        details,
+        modifier = Modifier.padding(10.dp)
+    )
+}
+
+@Composable
+fun MovieDetailsScreen(
+    movieDetailsViewState: MovieDetailsViewState,
+    modifier: Modifier = Modifier,
+) {
+    LazyColumn {
+        item {
+            MovieDetailsAll(
+                movieDetailsViewState = movieDetailsViewState,
+                modifier = modifier
+            )
+        }
+    }
+}
+
+@Composable
+fun MovieDetailsAll(
+    movieDetailsViewState: MovieDetailsViewState,
+    modifier: Modifier = Modifier,
+) {
+    MovieDetailsBanner(movieDetailsViewState = movieDetailsViewState)
+    Spacer(modifier = modifier.size(10.dp))
+    MovieDetailsOverview(movieDetailsViewState = movieDetailsViewState)
+    Spacer(modifier = Modifier.size(10.dp))
+    MovieDetailsCast(movieDetailsViewState = movieDetailsViewState)
+}
+
+@Composable
+fun MovieDetailsBanner(
+    movieDetailsViewState: MovieDetailsViewState,
+    modifier: Modifier = Modifier,
+) {
+    Box(modifier = modifier
+        .fillMaxWidth()
+        .height(400.dp)
+    ) {
+        AsyncImage(
+            model = "${movieDetailsViewState.imageUrl}",
+            contentDescription = null,
+            modifier = Modifier.fillMaxSize(),
+            contentScale = ContentScale.Crop
+        )
+        Column(
+            modifier = Modifier.fillMaxSize(),
+            verticalArrangement = Arrangement.Bottom
+        ) {
+            Row(
+                modifier = Modifier.fillMaxWidth(),
+                verticalAlignment = Alignment.CenterVertically
+            ) {
+                UserScoreProgressBar(modifier = Modifier.padding(start = 10.dp),
+                    percentage = movieDetailsViewState.voteAverage)
+                Text(
+                    text = "User Score",
+                    color = Color.White,
+                    modifier = Modifier.padding(5.dp),
+                    fontWeight = FontWeight.Bold,
+                )
+            }
+            Spacer(modifier = Modifier.size(10.dp))
+            Text(
+                modifier = Modifier.padding(start = 10.dp),
+                text = "${movieDetailsViewState.title}",
+                color = Color.White,
+                fontWeight = FontWeight.Bold,
+            )
+            Spacer(modifier = Modifier.size(10.dp))
+            FavoriteButton(isFavorite = movieDetailsViewState.isFavorite) {
+                !movieDetailsViewState.isFavorite
```

Comment:
you're just calling negation on that property, you should add onFavoriteClick parameter to MovieDetailsBanner function and call it with this negation